File: Communication entre processus/exo4.2.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main() 
{
    
    char *env_val = getenv("MON_TUBE");
    
    if (env_val == NULL) return 1;

    int fd_lecture = atoi(env_val);
    char tampon[100];

    read(fd_lecture, tampon, 100);
    printf("Fils (via ENV) a reçu : %s\n", tampon);

    close(fd_lecture);
    return 0;
}

File: Communication entre processus/exo4.2.py
import os
import sys

def main():
    fd_str = os.environ.get("MON_TUBE")
    if fd_str is None:
        sys.exit(1)
        
    fd_lecture = int(fd_str)
    
    tampon = os.read(fd_lecture, 100)
    print(f"Fils (env) a reçu : {tampon.decode().strip(chr(0))}")

    os.close(fd_lecture)

if __name__ == "__main__":
    main()